### ISSUE TITLE
Please provide Windows ARM64 wheels

### ISSUE THREAD

**Issue by nanonyme** [python-cffi/cffi#109]
There is now native ARM64 Python since release 3.12. many central projects including this one do not provide wheels for this platform. Please start building wheels also for Windows ARM64 when convenient.

**Comment by ofek** [comment]
Yes this would be fantastic, we've been getting issues where users can no longer install our packages.

**Comment by nanonyme** [comment]
It's possible this is pending GitHub ARM64 Windows project like for many other Python projects. That in term is most likely pending Windows Server 2025 release which is the first release to support ARM64.

**Comment by ofek** [comment]
Cross compilation is possible, for example cibuildwheel already supports doing that for Windows ARM.

**Comment by mattip** [comment]
Native Arm64 windows github action runners [are in beta](https://github.blog/changelog/2024-06-03-actions-arm-based-linux-and-windows-runners-are-now-in-public-beta/). I am not aware of an easy way to get docker or qemu to run windows-arm64 on a github CI runner for testing. 

> users can no longer install our packages

You imply that once they could and now they can't. Is that what you meant?

> Cross compilation is possible

Yes, but that does not solve the testing problem. I would be hesitant to release a package like CFFI without extensively testing that it works on a new platform. This is also the approach NumPy has taken: it will not release windows-arm64 wheels without testing. Here is the relevant [comment chain](https://github.com/numpy/numpy/issues/22530#issuecomment-2129519734) on the issue.

**Comment by mattip** [comment]
A quick search on another base package in the python extensions multiverse, cryptography, resulted in a similar approach: [no wheels without testing](https://github.com/pyca/cryptography/issues/11199).

**Comment by nanonyme** [comment]
> Native Arm64 windows github action runners [are in beta](https://github.blog/changelog/2024-06-03-actions-arm-based-linux-and-windows-runners-are-now-in-public-beta/). I am not aware of an easy way to get docker or qemu to run windows-arm64 on a github CI runner for testing.
> 

Yeah, I've been digging around and the reason it's only in beta even though Microsoft is heavily pushing ARM64 most likely is because there is no existing non-beta Windows Server that can run on ARM64. It's no doubt going to be released any year now.

**Comment by nanonyme** [comment]
> Yes, but that does not solve the testing problem. I would be hesitant to release a package like CFFI without extensively testing that it works on a new platform. This is also the approach NumPy has taken: it will not release windows-arm64 wheels without testing. Here is the relevant [comment chain](https://github.com/numpy/numpy/issues/22530#issuecomment-2129519734) on the issue.

I wholeheartedly agree. If you crosscompile, you need *some* way to test. Where I work this meant completely different system to where build actually happened. Windows ARM64 isn't really yet in great shape. The issue was filed in good faith requesting this *eventually* since I didn't find existing issue.

**Comment by mattip** [comment]
> The issue was filed in good faith

I am not really a core dev here, but thanks for the clarification. Bootstrapping a new architecture, especially for a platform as popular as windows, is going to take a while. 

I think CPython could only release an experimental version because they (a) run a buildbot farm of distributed machines rather than being dependent on centralized CI. This of course has its own security implications, and requires a larger effort. And (b) CPython has the support of [the linaro Windows On Arm team](https://linaro.atlassian.net/wiki/spaces/WOAR/overview) who run the [CPython buildbot](https://buildbot.python.org/#/workers/45). Maybe someone knows how to talk to them and get some core projects CI runners?

**Comment by ncoghlan** [comment]
Unfortunately, the trust management issues around the core CPython buildbot fleet mean that even offering other projects intermittent access isn't feasible, let alone routine access for artifact creation.

I'm sure MS would like their Copilot+ PCs to be practical platforms for AI *development* in addition to AI consumption, so I'll see if anyone in the core dev team has GitHub contacts to talk to about community orgs getting access to the ARM64 beta runners.

**Comment by localden** [comment]
@ncoghlan let me know if you don't have any direct contacts and I can likely help connect. We've been looking at deploying some of the auth stack pieces on Windows on ARM (`win-arm64`) and the `cffi` dependency blocks us from successfully rolling the code out on the platform.

**Comment by nitzmahone** [comment]
> It's possible this is pending GitHub ARM64 Windows project like for many other Python projects

Yep, exactly.

I can't justify the time to create and maintain bespoke WoA GHA runner infrastructure. Windows aarch64 wheels are unlikely to happen anytime soon unless Microsoft does GH-hosted WoA runners like they recently did for MacOS on Apple Silicon. Python wheels with cross-compiled statically-linked extensions are fraught with peril and must be thoroughly tested under their native arches. That said, I can't justify the time to create and maintain a bespoke WoA runner pool...

**Comment by tdunkleArm** [comment]
Hello everybody, I wanted to bring to attention that Microsoft published [a blog post](https://blogs.windows.com/windowsdeveloper/2025/04/14/github-actions-now-supports-windows-on-arm-runners-for-all-public-repos/) yesterday about GitHub now hosting WoA runners that are available to all public repos.

@nitzmahone I wanted to put this topic back on your radar. Are you still willing to add CI and wheel support now that the runner should be available to this project?
